### PR TITLE
refactor(ng-dev): remove minimist usage

### PR DESCRIPTION
**Pull request by alan-agius4**
Use `yargs` since it's the preferred parser.

### CODE REVIEW

File: ng-dev/utils/testing/virtual-git-client.ts [hunk 1]
```diff
@@ -99,8 +99,14 @@ export class VirtualGitClient extends AuthenticatedGitClient {
 
   /** Handler for the `git push` command. */
   private _push(args: string[]) {
-    const [repoUrl, refspec] = parseArgs(args, {boolean: ['q']})._;
-    const ref = this._unwrapRefspec(refspec);
+    const [repoUrl, refspec] = yargs(args)
```
Comment (devversion) [inline ng-dev/utils/testing/virtual-git-client.ts:102]: Is there a way to manage `repoUrl` and `refspec` as "positional" here? that would save us from adding the type casts.

**Review by devversion — APPROVED** [review]
👍 Minimist is less verbose for the tests.. but Yargs is definitely the better choice.

File: ng-dev/utils/testing/virtual-git-client.ts (hunk 1 above)
Comment (alan-agius4) [reply ng-dev/utils/testing/virtual-git-client.ts:102]: You'd need to define commands. 
Example: https://github.com/yargs/yargs/blob/master/docs/advanced.md#advanced-topics

**Comment by alan-agius4** [comment]
> Minimist is less verbose for the tests

Agreed, but since yargs is already a dependency might as well use that.

File: ng-dev/utils/testing/virtual-git-client.ts (hunk 1 above)
Comment (devversion) [reply ng-dev/utils/testing/virtual-git-client.ts:102]: which I guess could make sense as it's basically mocking a command, but I'm fine either way.

Comment (alan-agius4) [reply ng-dev/utils/testing/virtual-git-client.ts:102]: I can get back to replacing this next week.

**Review by josephperrott — APPROVED** [review]

File: ng-dev/utils/testing/virtual-git-client.ts [hunk 2]
```diff
@@ -112,15 +117,17 @@ export class VirtualGitClient extends AuthenticatedGitClient {
     if (existingPush !== undefined) {
       existingPush.head = pushedHead;
     } else {
-      this.pushed.push({remote: {repoUrl, name}, head: pushedHead});
+      this.pushed.push({remote: {repoUrl: repoUrl, name}, head: pushedHead});
```
Comment (devversion) [inline ng-dev/utils/testing/virtual-git-client.ts:120]:
```suggestion
      this.pushed.push({remote: {repoUrl, name}, head: pushedHead});
```

File: ng-dev/utils/testing/virtual-git-client.ts [hunk 3]
```diff
@@ -154,10 +166,19 @@ export class VirtualGitClient extends AuthenticatedGitClient {
...
+      target,
+    } = yargs(rawArgs)
+      .command('$0 <target>', false)
+      .positional('target', {demandOption: true, type: 'string'})
+      .option('detach', {boolean: true})
+      .option('q', {boolean: true})
+      .option('B', {boolean: true})
+      .parseSync();
+
+    //const target = targets[0];
```
Comment (devversion) [inline ng-dev/utils/testing/virtual-git-client.ts:181]:
```suggestion
```

**Review by devversion — APPROVED** [review]

**Pull request closed without merging**